packages/taiko-client/driver/chain_syncer/blob/syncer.go

```diff
@@ -315,25 +324,150 @@ func (s *Syncer) checkLastVerifiedBlockMismatch(ctx context.Context) (*rpc.Reorg
 		return nil, fmt.Errorf("failed to fetch L2 header by number: %w", err)
 	}
 
+	// If the last verified block hash matches the L2 EE block hash, we skip the check.
 	if header.Hash() == lastVerifiedBlockHash {
 		return reorgCheckResult, nil
 	}
 
-	genesisL1Header, err := s.rpc.GetGenesisL1Header(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch genesis L1 header: %w", err)
+	// If the L2 chain is on Pacaya fork.
+	for {
+		if lastVerifiedBlockID >= s.rpc.PacayaClients.ForkHeight {
```

Comment:
```suggestion
	for lastVerifiedBlockID >= s.rpc.PacayaClients.ForkHeight {
```

Reply:
And delete `else` and `continue` at the end of the loop.

Reply:
changed: https://github.com/taikoxyz/taiko-mono/pull/19162/commits/79285f8e9cafc15b837ea92e500c3ac6070d78e0